In [3]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [6]:
df = pd.read_csv("Dataset2.csv")
df.head()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,125,33.6,0.627,50,1
1,1,85,66,29,95,26.6,0.351,31,0
2,8,183,64,32,180,28.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


In [7]:
X = df.drop('Outcome',axis=1)
y = df['Outcome']

In [8]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [9]:
model = LogisticRegression(max_iter=200)
model.fit(X_train,y_train)

LogisticRegression(max_iter=200)

In [10]:
y_pred = model.predict(X_test)
print(y_pred)

[1 1 1 0 0 0 0 1 0 1 0 1 0 1]


In [12]:
acc = accuracy_score(y_test, y_pred)
conf = confusion_matrix(y_test, y_pred)
rep = classification_report(y_test, y_pred)

print("Accuracy:", acc)
print("Confusion Matrix:\n", conf)
print("Classification Report:\n", rep)

Accuracy: 0.9285714285714286
Confusion Matrix:
 [[7 1]
 [0 6]]
Classification Report:
               precision    recall  f1-score   support

           0       1.00      0.88      0.93         8
           1       0.86      1.00      0.92         6

    accuracy                           0.93        14
   macro avg       0.93      0.94      0.93        14
weighted avg       0.94      0.93      0.93        14



Confusion Matrix Visualization
           Predicted
           0       1
Actual 0  [7]     [1]
         TN       FP
Actual 1  [0]     [6]
         FN       TP

Legend:

TN (True Negative) = 7 → Correctly predicted class 0

FP (False Positive) = 1 → Predicted 1 but actually 0

FN (False Negative) = 0 → Predicted 0 but actually 1

TP (True Positive) = 6 → Correctly predicted class 1

Precision & Recall
Class	Precision	Recall	F1-Score	What it Means
0	1.00	0.88	0.93	When predicting 0, always correct; caught 88% of actual 0s
1	0.86	1.00	0.92	When predicting 1, correct 86% of time; caught all actual 1s
Visual Diagram (Easy Way to Remember)
           Predicted
           0        1
Actual 0  ✅7      ❌1
Actual 1  ❌0      ✅6

Precision 0: 1.0    Precision 1: 0.86
Recall    0: 0.88   Recall 1: 1.0

✅ = Correct prediction

❌ = Mistake

Key insight:

Most errors come from class 0 being mistaken as 1 once.

Model is strong for class 1 detection (recall = 1).